python/sidecar/projects.py

```diff
@@ -106,7 +107,62 @@ def delete_project(user_id: str, project_id: str) -> None:
         pass
 
 
-def get_project_files(user_id: str, project_id: str) -> list:
+def get_project_files(user_id: str, project_id: str) -> ProjectFileTreeResponse:
+    """
+    Get the files in a project as a tree structure.
+
+    Returns
+    -------
+    ProjectFileTreeResponse
+        A tree structure of the files in a project.
+    """
     project_path = get_project_path(user_id, project_id)
-    filepaths = list(project_path.glob("**/*"))
-    return filepaths
+    project_name = get_project_name(user_id, project_id)
+
+    if not project_path.exists():
+        raise FileNotFoundError(f"Project {project_id} does not exist")
+
+    contents = []
+    for obj in project_path.rglob("*"):
+        relpath = obj.relative_to(project_path)
+
+        # Skip hidden files and directories
+        if relpath.name.startswith(".") or obj.name.startswith("."):
+            continue
+
+        # files in subdirectories
+        if obj.is_dir():
```

Comment:
> We can drop isFile and use the undefined children to detect if the entry is a file or folder.

@cguedes can you clarify that piece from the issue?

With the current code, this would be the structure for a nested directory:
```
[{
	"name": "empty",
	"path": "empty",
	"children": []
}, {
	"name": "uploads",
	"path": "uploads",
	"children": [{
		"name": "file2.pdf",
		"path": "uploads/file2.pdf",
		"file_extension": ".pdf"
	}]
}, {
	"name": "nested",
	"path": "nested",
	"children": [{
		"name": "file3.pdf",
		"path": "nested/folder/file3.pdf",
		"file_extension": ".pdf"
	}]
}, {
	"name": "file1.refstudio",
	"path": "file1.refstudio",
	"file_extension": ".refstudio"
}, {
	"name": "folder",
	"path": "nested/folder",
	"children": [{
		"name": "file3.pdf",
		"path": "nested/folder/file3.pdf",
		"file_extension": ".pdf"
	}]
}]
```

Reply:
No I would expect a tree of **File** / **Directory** entries.

For this scenario it would be:

Note the `folder` directory is sibling of `file3.pdf` and that `file4.pdf` is "inside" the `folder` as children.


```json
[
  {
    "name": "empty",
    "path": "empty",
    "children": []
  },
  {
    "name": "uploads",
    "path": "uploads",
    "children": [
      {
        "name": "file2.pdf",
        "path": "uploads/file2.pdf",
        "file_extension": ".pdf"
      }
    ]
  },
  {
    "name": "nested",
    "path": "nested",
    "children": [
      {
        "name": "file3.pdf",
        "path": "nested/file3.pdf",
        "file_extension": ".pdf"
      },
      {
        "name": "folder",
        "path": "nested/folder",
        "children": [
          {
            "name": "file4.pdf",
            "path": "nested/folder/file4.pdf",
            "file_extension": ".pdf"
          }
        ]
      }
    ]
  },
  {
    "name": "file1.refstudio",
    "path": "file1.refstudio",
    "file_extension": ".refstudio"
  }
]
```

Reply:
The output is still wrong 
<img width="918" alt="image" src="https://github.com/refstudio/refstudio/assets/174127/336c02a7-8951-4981-9266-e1b36056d168">